File: app/src/main/java/me/dev4vin/samplepromiseapp/ui/views/LoadableView.java
package me.dev4vin.samplepromiseapp.ui.views;

import android.view.View;

import me.yoctopus.model.Viewable;

public class LoadableView implements Viewable {
  private String status;

  public LoadableView(String status) {
    this.status = status;
  }

  @Override
  public int layout() {
    return 0;
  }

  @Override
  public void init(View view) {

  }

  @Override
  public void bind(View view) {

  }

  @Override
  public void index(int index) {

  }

  @Override
  public int index() {
    return 0;
  }
}

File: app/src/main/java/promise/app/ui/views/LoadableView.kt
package promise.app.ui.views

import android.view.View
import androidx.appcompat.widget.AppCompatTextView
import promise.app.R

import promise.model.Viewable

class LoadableView(private val status: String) : Viewable {

  private lateinit var textView: AppCompatTextView

  override fun layout(): Int = R.layout.loadable_view_layout

  override fun init(view: View) {
    textView = view.findViewById(R.id.loading_text_textView)
  }

  override fun bind(view: View) {
    textView.text = status
  }

  override fun index(index: Int) {

  }

  override fun index(): Int = 0
}
